### PR TITLE
fix pods pending but vcjob is running

### PR DESCRIPTION
**Pull request by xiefan-github**
[issues #2884](https://github.com/volcano-sh/volcano/issues/2884)

### CODE REVIEW

File: pkg/controllers/job/state/running.go [hunk 1]
```diff
@@ -91,6 +91,10 @@ func (ps *runningState) Execute(action v1alpha1.Action) error {
 				}
 				return true
 			}
+			if status.Pending == jobReplicas {
```
Comment (hwdef) [inline pkg/controllers/job/state/running.go:94]: should it be `status.Pending > jobReplicas - totalMinavailable`

Comment (xiefan-github) [reply pkg/controllers/job/state/running.go:94]: yes, you're right, it's more accurate, I'll fix it

Comment (xiefan-github) [reply pkg/controllers/job/state/running.go]: I have fixed it

**Review by hwdef — COMMENTED** [review]
/lgtm

**Comment by lowang-bh** [comment]
could you add more testcase to cover your code, pls?

**Comment by xiefan-github** [comment]
> could you add more testcase to cover your code, pls?

Of course, I'd be happy to write some test cases，give me some time

**Comment by xiefan-github** [comment]
> could you add more testcase to cover your code, pls?

could you review my testcase，Thank you

File: pkg/controllers/job/state/running.go (hunk 1 above)
Comment (xiefan-github) [reply pkg/controllers/job/state/running.go]: I have a confusion why totalMinavailable  can not be equal to Job.Spec.MinAvailable? #2895

**Comment by lowang-bh** [comment]
> > could you add more testcase to cover your code, pls?
> 
> could you review my testcase，Thank you

It's ok to me. @wangyang0616 @hzxuzhonghu plz also help to review. thanks.

**Review by hwdef — COMMENTED** [review]
/lgtm

**Comment by wangyang0616** [comment]
Thanks for your contribution. It's ok to me.

Please help trigger CI. Thanks!  @william-wang

File: pkg/controllers/job/state/running.go (hunk 1 above)
Comment (hwdef) [reply pkg/controllers/job/state/running.go]: @xiefan-github 
When I first raised the question, I was thinking that if the number of pending pods is lower than minavailable, the pod should be in the pending state.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**